akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -220,4 +220,11 @@ object Sink extends SinkApply {
     new Sink(new AcknowledgeSink(bufferSize, DefaultAttributes.acknowledgeSink, shape("AcknowledgeSink"), timeout))
   }
 
+  /**
+   * Crates a `Sink` that is materialized into `Source` to connect with another flow.
+   * This sink gets backpressure from second flow and does not have internal buffer.
+   */
+  def pipedSink[T](): Sink[T, Source[T, Unit]] =
+    Sink.publisher[T].mapMaterializedValue(Source(_)).named("PipedSink")
```

Comment:
> The streams philosophy (as the API is currently) **favors** encoding dependencies in your modules directly

Emphasis mine, I never said that you cannot use it any other way.

> **While this ideal cannot be met in 100% of the cases**, in general the unit of modularity ... 

Emphasis mine, I never said that it is the only way. Nevertheless, this is the more natural way given the API, since you cannot split up a Flow to a Source and Sink if a pair of functions require those. 

You cannot cut up a Flow (like with using the Sink.source here) without actually materializing one half of it. That is what I said, please read it again. there is no "talking past" on my side. We provided the convenience method for prematerializing stuff, but what I cannot do is to throw out the current materializer and provide means for detaching part of the graph in different DSL elements but still materializing together, and still maintain reusability. These are conflicting goals.